src/KeyPair.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Tighten\SolanaPhpSdk;
+
+use Tighten\SolanaPhpSdk\Exceptions\TodoException;
+use Tighten\SolanaPhpSdk\Util\Ed25519Keypair;
+
+/**
+ * An account keypair used for signing transactions.
+ */
+class KeyPair
+{
+    protected Ed25519Keypair $_keypair;
+
+    /**
+     * @param $secretKey
+     */
+    public function __construct(?Ed25519Keypair $keypair = null)
+    {
+        $this->_keypair = $keypair ?? Ed25519Keypair::generate();
+    }
+
+    /**
+     * Generate a new random keypair
+     */
+    static function generate(): KeyPair
```

Comment:
I *think* you can type `static`.. not a huge difference but I think it's a bit clearer.

```suggestion
    static function generate(): static
```

Reply:
Love it!! Agreed cleaner, however, that is supported in PHP 8.0+. We probably should keep the min as low as possible for now?